lib/payment_dupes.rb

```diff
@@ -43,36 +47,65 @@ def self.can_copy_designation?(src, target, designations_to_become_comments)
         true
     end
 
-    def self.copy_comment(source, target)
+    def self.copy_comment(source, target, designations_to_become_comments)
         return true if source.donation.comment.blank?
         return true if target.donation.comment.present? && (source.donation.comment.blank? || target.donation.comment == source.donation.comment)
-        return false if target.donation.comment.present?
+        if target.donation.comment.present? 
+            if designations_to_become_comments.any? { |d| target.donation.comment.include?(d) }
+                designations_already_copied_to_comment = designations_to_become_comments.select { |d| target.donation.comment.include?(d) }
+                comment = target.donation.comment
+                designations_already_copied_to_comment.each do |d|
+                    comment = target.donation.comment.gsub(" \nDesignation: #{d}", "")
+                    comment = target.donation.comment.gsub("Designation: #{d}", "")
+                end
+                return true if (source.donation.comment.blank? || comment == source.donation.comment)
+            else
+                return false
+            end
+        end
         target.donation.comment = source.donation.comment
         target.donation.save!
     end
 
-    def self.can_copy_comment?(source, target)
+    def self.can_copy_comment?(source, target, designations_to_become_comments)
```

Comment:
Where is "designations_to_become_comments" coming from originally?

Reply:
https://github.com/CommitChange/houdini/pull/268/files#diff-0dbca3d4691fa54ac9678cda39b5ec2e9d18eed195ffa425fe2adbed140d82d5R88

Reply:
Could you explain it in another way? `designations_to_become_comments` is a list of designations that we're moving to a comment because they are conflicting with the online payment designation but shouldn't be a problem, but because there was a bug in the payments deletion, the donations already had the designation as a comment.

Reply:
Yes, take a look at step 8: https://github.com/CommitChange/tix/issues/3615#issuecomment-1034458827